### DISCUSSION TITLE
dzsave of large, high resolution pdf (dpi 4800) produces only empty tiles

### DISCUSSION THREAD

**Discussion by liveman1** [libvips/libvips#2297] [Q&A]
Hi,

I would like to create a DZI from a large PDF with a higher resolution (dpi) using following commando:
vips-dev-8.11\bin>vips.exe dzsave sample\test.pdf[dpi=4800] test.zip --layout dz --tile-size 256 --suffix .jpg[Q=70]

Unfortunately, I get only empty tiles.

Everything works perfectly nice when setting a lower resolution, eg dpi=2400.

Do you have any idea what I am doing wrong?

Thanks a lot!

**Comment by jcupitt** [comment]
Hello again, I would guess one axis is going over 32,000 pixels, just as the SVG did. I imagine both libraries are using Cairo for rendering. You can test like this:

```
$ vipsheader nipguide.pdf[dpi=2500]
nipguide.pdf: 20669x29232 uchar, 4 bands, srgb, pdfload
```

And that works, but:

```
$ vipsheader nipguide.pdf[dpi=5000]
nipguide.pdf: 41339x58465 uchar, 4 bands, srgb, pdfload
```

Fails to create a deepzoom for me (all tiles white).

libvips also supports PDFium as a PDF renderer, you could try that. It would involve rebuilding the library, unfortunately, which is not easy on Windows.

**Comment by liveman1** [comment]
Thanks a lot jcupitt!
Why is there a limit of 32000 pixels?
Is there any other way to do this using libvips? E.g. some command to crop a high resolution svg in smaller pieces and then building the DZI?

**Comment by jcupitt** [comment]
Cairo (the rendering library used by rsvg) represents pixel coordinates as 16.16 bit fixed point, so coordinates have the range -32768 to +32767. Yes, libvips does the scaling-translating-cropping trick internally, and it used to work, but it seems that librsvg changed that behaviour recently and it's no longer allowed.

I'd guess poppler is also using Cairo at some point and has the same limit.
